ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -522,23 +527,54 @@ def test_userroot_ldif_files_ownership(self):
                                  '--uninstall',
                                  '-U'])
 
+        # get default umask
+        default_umask = self.master.run_command(['umask'])
+
+        # set umask to 077 just to check if restore success
+        self.master.run_command('echo "umask 0077" >> ~/.bashrc')
+        result = self.master.run_command(['umask'])
+        assert '0077' in result.stdout_text
+
         dirman_password = self.master.config.dirman_password
-        self.master.run_command(['ipa-restore', backup_path],
-                                stdin_text=dirman_password + '\nyes')
+        result = self.master.run_command(['ipa-restore', backup_path],
+                                         stdin_text=dirman_password + '\nyes')
+        assert 'Temporary setting umask to 022' in result.stderr_text
+
+        # check if umask reset to 077 after restore
+        result = self.master.run_command(['umask'])
+        assert '0077' in result.stdout_text
+
+        # restore default umask
+        cmd = 'echo "umask {}" >> ~/.bashrc'.format(default_umask)
+        self.master.run_command(cmd)
```

Comment:
While this will restore the value it extends the bashrc so theoretically after many runs the same VM could have multiple setting of umask. The last one wins but this seems messy. Can you not instead save a copy of bashrc and restore that copy when the test finishes? You probably want to see if this cleanup can be done during test teardown so that if the test fails it will always be restored to the previous value.